control_protocol.md

```diff
@@ -0,0 +1,359 @@
+    C0([nS.COORDINATOR DEALER]) == "V|nR.recipient|nS.CA|H|Content" ==> R0
+    R0[receive] == "iA|V|nR.recipient|nS.CA|H|Content" ==> CnS{nS == N1?}
+    CnS-->|no| RemIdent
+    CnS-->|yes| Clocal{CA in <br>local directory?}
+    Clocal -->|yes| CidKnown{iA is CA's identity<br> in directory?}
+    CidKnown -->|yes| RemIdent
+    Clocal -.->|no| E1[ERROR: Sender unknown] ==>|"iA|V|nS.CA|N1.COORDINATOR|H|ERROR: Sender unknown"| S
+    S[send] ==> WA([N1.CA DEALER])
+    CidKnown -.->|no| E2[ERROR: Name and identity do not match]==>|"iA|V|nS.CA|N1.COORDINATOR|H|ERROR: Name and identity do not match"| S
+    RemIdent[remove sender identity] == "V|nR.recipient|nS.CA|H|Content" ==> CnR
+    CnR -- "is None" --> Local
+    CnR{nR?} -- "== N1"--> Local
+    Local{recipient<br>==<br>COORDINATOR?} -- "yes" --> Self[Message for Co1<br> itself]
+    Self == "V|nR.recipient|nS.CA|H|Content" ==> SC([Co1 Message handling])
+    Local -- "no" --> Local2a{recipient in directory?}
+    Local2a -->|yes, with Identity iB| Local2
+    Local2[add recipient identity iB] == "iB|V|nR.recipient|nS.CA|H|Content" ==> R1[send]
+    R1 == "V|nR.recipient|nS.CA|H|Content" ==> W1([Wire to N1.recipient DEALER])
+    Local2a -.->|no| E3[ERROR recipient unknown<br>send Error to original sender] ==>|"V|nS.CA|N1.COORDINATOR|H|ERROR N1.recipient is unknown"|CnR
+    CnR -- "== N2" --> Keep
+    Keep[send to N2.COORDINATOR] == "V|nR.recipient|nS.CA|H|Content" ==> R2[send]
+    R2 == "V|nR.recipient|nS.CA|H|Content" ==> W2([Wire to N2.COORDINATOR ROUTER])
+    subgraph Co1 ROUTER socket
+        R0
+    end
+    subgraph Co1 ROUTER socket
+        R1
+        S
+    end
+    subgraph Co1 DEALER socket <br>to N2.COORDINATOR
+        R2
+    end
+:::
+
+
+#### Coordinator coordination
+
+Each Coordinator shall keep an up to date global {ref}`directory` with the names of all Components in the Network.
+For this, Coordinators shall tell each other regarding signing in and signing out Components and Coordinators.
+Coordinators shall send on request the Names of their local directory, or of their global directory, depending on the request type.
+
+For the format of the Messages, see {ref}`message-layer`.
+
+
+##### Coordinator sign in
+
+A Coordinator `Co1`joining a network follows a few steps:
+1. It signs in to one Coordinator `Co2` of the Network.
+2. It sends a CO_TELL_ALL message to `Co2`, to tell all other Coordinators about `Co1`s address.
+3. `Co2` tells all the Coordinators signed in (`Co3`, `Co4`...) about `Co1` with a CO_NEW message.
+4. These other Coordinators (`Co3`, `Co4`...) sign in to `Co1`.
+5. All Coordinators are connected to all others.
+
+Two Coordinators shall follow a more thorough sign in/sign out procedure, than Components (address is for example host an port).
+The sign in might happen because of a CO_NEW message arrived or at startup.
+The sign out might happen because the Coordinator shuts down.
```

Comment:
```suggestion
Two Coordinators shall follow a more thorough sign-in/sign-out procedure than Components (address is for example host and port).
The sign-in might happen because of a CO_NEW message arrived or at startup.
The sign-out might happen because the Coordinator shuts down.
```